docs/examples/tutorial/cython_tutorial/primes_cpp.py

```diff
@@ -0,0 +1,22 @@
+# distutils: language=c++
+
+import cython
+from cython.cimports.libcpp.vector import vector
+
+def primes(nb_primes: cython.uint):
+    i: cython.int
+    p: vector[cython.int]
+    p.reserve(nb_primes)  # allocate memory for 'nb_primes' elements.
+
+    n: cint = 2
+    while p.size() < nb_primes:  # size() for vectors is similar to len()
+        for i in p:
+            if n % i == 0:
+                break
+        else:
+            p.push_back(n)  # push_back is similar to append()
+        n += 1
+
+    # Vectors are automatically converted to Python
+    # lists when converted to Python objects.
+    return p
```

Comment:
I don't remember, was this supposed to be able to run in Python or not?

Reply:
It's not, and [here](https://github.com/cython/cython/pull/4226/files#diff-b10a6fc196b2dff317a7a7361bbf9c0f6da041f17e59dbf12c62ff17a5e797edR427-R436)'s warning about it

Reply:
No. It only needs to runnable when compiled in Cython and otherwise compilable/importable in plain Python. (There is code in `runtests.py` now that makes sure that the import does not fail, and not more than that.)

Reply:
Ah, what confused me was the usage of pure Python typehints as opposed to Pyrex style definitions.